### PR TITLE
Source map

### PR DESCRIPTION
**Pull request by katatsu12**

### CODE REVIEW

**Comment by katatsu12** [comment]
```
hash_result = `node /app/assets/javascripts/parser.js map=#{'./path/to/map'} line=#{n} column=#{n}`
```

**Comment by katatsu12** [comment]
Screenshot from 2020-04-26 12-42-01

File: app/services/events/resolve_source_code_service.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    end
+
+    def save_source_map
+      download.is_a?(File) ? download : IO.copy_stream(download, path)
+    end
+
+    # rubocop:disable Security/Open
+    def download
+      @download ||= begin
+        open(source_map_url)
```
Comment (Alexey-Lukin) [inline app/services/events/resolve_source_code_service.rb:33]: !

File: lib/javascript/parser.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+var args = {  }
+var fs = require('fs');
+var sourceMap = require('source-map');
+
+process.argv.slice(2).forEach(arg => {
+  var [key,value] = arg.split(':')
+  args[key] = value || true
```
Comment (Alexey-Lukin) [inline lib/javascript/parser.js:7]: always true?

File: app/services/events/create_service.rb [hunk 3]
```diff
@@ -3,14 +3,25 @@
...
+    resolve_source_code if resolve_source_code?
+    event.save
     notify if notify?
     [{ message: I18n.t('api.event_captured') }, 201]
   end
 
   private
 
+  def resolve_source_code?
+    event.framework == 'browser-js' && event.backtrace.present?
```
Comment (viktor-shmigol) [inline app/services/events/create_service.rb:17]: move to constant

**Pull request merged**